parsers/jscript.c

```diff
@@ -1913,6 +1926,7 @@ static bool parseMethods (tokenInfo *const token, const tokenInfo *const class,
 	findCmdTerm (token, false, false);
 
 cleanUp:
+	token->scopeKind = saveScopeKind;
 	vStringCopy (token->scope, saveScope);
```

Comment:
How about introducing something like `fillTokenScope(tokenInfo *token, vString *saveScope, int saveScopeKind);`?

Reply:
`setScope` may be better as the name of the function.